src/website/views/components/forms/text-input/_examples.pug

```diff
@@ -53,20 +54,56 @@ h3 Disabled
         <input type="text" class="chi-input" value="Sample Text" id="example__disabled" disabled>
       </div>
 
-h3 Invalid
-p.-text Text inputs support Chi's semantic class modifier <code>-danger</code> for indicating an invalid input.
+h3 Text Input helper message
 .example.-mb--3
-  .-p--3
-    .chi-form__item(style="max-width:20rem")
-      chi-label(for="example__danger") Label
-      chi-text-input(
-        id="example__danger"
-        state="danger"
-        icon-right='circle-warning'
-        icon-right-color='danger'
-        value="Sample Text"
+  .chi-grid.-p--3
+    .chi-col.-w--12.-w-md--6
+      .chi-form__item(style="max-width: 20rem;")
+        chi-label(for="example-message__success") Label
+        chi-text-input(
+          id="example-message__success"
+          state="success"
+          icon-right='check'
+          value="Sample Text"
+        )
+          div(slot="chi-form__helper-message")
+            | Helper text and icon color is inherited
+    .chi-col.-w--12.-w-md--6.-pt--3.-pt-md--0
+      .chi-form__item(style="max-width: 20rem;")
+        chi-label(for="example-message__warning") Label
+        chi-text-input(
+          id="example-message__warning"
+          state="warning"
+          icon-right='warning'
+          value="Sample Text"
+        )
+          div(slot="chi-form__helper-message")
+            | Helper text and icon color is inherited
+    .chi-col.-w--12.-w-md--6.-pt--3
+      .chi-form__item(style="max-width: 20rem;")
+        chi-label(for="example__danger") Label
+        chi-text-input(
+          id="example__danger"
+          state="danger"
+          icon-right='circle-warning'
+          value="Sample Text"
         )
-      .chi-label.-status.-danger This is an invalid label
+          div(slot="chi-form__helper-message")
```

Comment:
Another good example to see here would be no semantic class defined. This would be for the help text use case.